src/js/jsx/sections/export/ExportAllPanel.jsx

```diff
@@ -0,0 +1,226 @@
+                        {assetTitle}
+                    </div>
+                );
+            });
+
+            return (
+                <div className="exports-panel__layer-wrapper" >
+                    <CheckBox
+                        checked={layer.exportEnabled}
+                        onChange={this._handleLayerSelectedChanged}
+                        size="column-2" />
+                    <div className="exports-panel__layer-info">
+                        <div className="exports-panel__layer__name">
+                            {layer.name}
+                        </div>
+                        <div className="exports-panel__layer-assets">
+                            {assetsComponent}
+                        </div>
+                    </div>
+                </div>
+            );
+        }
+    });
+
+    /**
+     * The main "EXPORT" panel that provides a list of all configured assets.
+     * Allows layers to be enabled/disabled for export,
+     * and initiates the process to export all enabled-AND-configured assets
+     */
+    var ExportAllPanel = React.createClass({
+        mixins: [FluxMixin, StoreWatchMixin("export", "application", "document")],
+
+        propTypes: {
+            dismissDialog: React.PropTypes.func
+        },
+
+        getDefaultProps: function () {
+            return {
+                dismissDialog: _.identity
+            };
+        },
+
+        getInitialState: function () {
+            // the fresh state property allows assets to be determined as "stale"
+            // meaning that they have been exported previously, but not during this modal session
+            return {
+                fresh: true
+            };
+        },
+
+        getStateFromFlux: function () {
+            var flux = this.getFlux(),
+                document = flux.store("application").getCurrentDocument(),
+                documentExports = document && flux.store("export").getDocumentExports(document.id);
+
+            return {
+                document: document,
+                documentExports: documentExports
+            };
+        },
+
+        /**
+         * Export all assets for layers that have been enabled for export (via the checkboxes)
+         * @private
+         */
+        _exportAllAssets: function () {
+            this.setState({ fresh: false }, function () {
+                var exportActions = this.getFlux().actions.export,
+                    document = this.state.document;
+                return exportActions.setAllAssetsRequested(document).then(function () {
+                    return exportActions.exportAllAssets(document);
+                });
+            }.bind(this));
+        },
+
+        render: function () {
+            var document = this.state.document,
+                layerExportsMap = this.state.documentExports && this.state.documentExports.layerExportsMap,
+                freshState = this.state.fresh;
+
+            if (!document || !layerExportsMap) {
```

Comment:
There is no reason I can think of why there would not be a document at this point. The panel architecture guarantees this so that these checks don't need to be made repeatedly.